### ISSUE TITLE
Ferdora coreos

### ISSUE THREAD

**Issue by AlexStorm1313** [gsauthof/dracut-sshd#73]
Is it possible to add a release for Fedora coreos?

**Comment by gsauthof** [comment]
I'm not familiar with Fedora CoreOS. Especially I don't know its approach to packaging.

From a superficial understanding I gather that it's basically a static (or immutable) environment for containers.

Also, when I go to dracut-sshd's Copr settings page I don't see any mentioning of CoreOS.

**Comment by champtar** [comment]
Fedora CoreOS uses rpm-ostree, so it uses the regular rpms from Fedora to be built, with some requirements to keep in mind:
- binaries and assets go in /usr, %post can mutate what is in /usr, but it doesn't have access to /var or the modified /etc, this means that data migration logic must not be in the rpm scriptlet but as part of your program or as a Prestart
- /usr is read-only at runtime so you must not mutate assets or ...
- RW data go into /var
- if possible default config goes into /usr, /etc uses some overlay system, so if you modify a file you will keep the modified version, else you will get the new version
- by default initramfs is no_host_only and built without access to modified /etc

dracut-sshd needs access to /etc and to /root/.ssh/authorized_keys while building the initramfs, I think if you use rpm-ostree local initramfs generation (rpm-ostree initramfs --enable) it might work, not sure if it has access to /root

**Comment by AlexStorm1313** [comment]
Thanks for the informative reply. I've tried layering the Fedora package, this succeeds without errors. The problems start when generating the initramfs like @champtar mentioned there seem to be a problem with finding the ssh-keys to include with the initramfs. I think it is possible to make it work, find the correct keys etc. But for now I gave up and went with the officially supported clevis method.
